### PR TITLE
add string utility header

### PR DESCRIPTION
**Pull request by kungyao**

### CODE REVIEW

File: src/StringUtility.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2021, Group5 Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
```
Comment (john0312) [inline src/StringUtility.h:5]: Please use #define guard. See: https://google.github.io/styleguide/cppguide.html#The__define_Guard

File: src/StringUtility.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+#include <vector>
+#include <string>
+
+namespace StringUtility {
+	std::vector<std::string> split(const std::string& str, const std::string& del);
+	std::string SHA1code(const std::string& str);
+	std::string BASE64code(const std::string& str);
+};
```
Comment (john0312) [inline src/StringUtility.h:13]: Please add new line at the end.

File: src/StringUtility.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+#include <vector>
+#include <string>
+
+namespace StringUtility {
+	std::vector<std::string> split(const std::string& str, const std::string& del);
+	std::string SHA1code(const std::string& str);
+	std::string BASE64code(const std::string& str);
```
Comment (john0312) [inline src/StringUtility.h:12]: Is this encode or decode?

File: src/StringUtility.h [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+#include <vector>
+#include <string>
+
+namespace StringUtility {
+	std::vector<std::string> split(const std::string& str, const std::string& del);
+	std::string SHA1code(const std::string& str);
```
Comment (john0312) [inline src/StringUtility.h:11]: SHA1 is a hash, so it doesn't code. Consider using just "SHA1".

File: src/StringUtility.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Copyright (c) 2021, Group5 Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+#include <vector>
+#include <string>
+
+namespace StringUtility {
+	std::vector<std::string> split(const std::string& str, const std::string& del);
```
Comment (john0312) [inline src/StringUtility.h:10]: Please add function comments, see: https://google.github.io/styleguide/cppguide.html#Function_Comments

Comment (john0312) [reply src/StringUtility.h:10]: Also, it is recommended to add usage examples.

Comment (john0312) [reply src/StringUtility.h:10]: Also, consider using library for this, such as boost.

**Comment by john0312** [comment]
Hello @kuojohnny , please take a look at this commit. Thank you.

File: src/StringUtility.h (hunk 1 above)
Comment (kungyao) [reply src/StringUtility.h]: done

File: src/StringUtility.h (hunk 2 above)
Comment (kungyao) [reply src/StringUtility.h]: done

File: src/StringUtility.h (hunk 3 above)
Comment (kungyao) [reply src/StringUtility.h]: revise to encodeBASE64

File: src/StringUtility.h (hunk 4 above)
Comment (kungyao) [reply src/StringUtility.h]: done

File: src/StringUtility.h (hunk 5 above)
Comment (kungyao) [reply src/StringUtility.h]: The implement will use boost::split. Done.

**Review by john0312 — APPROVED** [review]

**Pull request merged**